web/frontend/src/api/navigation.ts

```diff
@@ -1,27 +1,32 @@
-import { Client, commonApi, navigationApi } from '@viamrobotics/sdk';
+import { get } from 'svelte/store';
+import { commonApi, navigationApi } from '@viamrobotics/sdk';
 import { grpc } from '@improbable-eng/grpc-web';
 import { rcLogConditionally } from '@/lib/log';
+import { client } from '@/stores/client';
 
 export type NavigationModes =
   | typeof navigationApi.Mode.MODE_MANUAL
   | typeof navigationApi.Mode.MODE_UNSPECIFIED
   | typeof navigationApi.Mode.MODE_WAYPOINT
 
-export const setMode = (client: Client, name: string, mode: NavigationModes) => {
+export type LngLat = { lng: number, lat: number }
+export type Waypoint = LngLat & { id: string }
+
+export const setMode = (name: string, mode: NavigationModes) => {
   const request = new navigationApi.SetModeRequest();
   request.setName(name);
   request.setMode(mode);
 
   rcLogConditionally(request);
 
   return new Promise((resolve, reject) => {
-    client.navigationService.setMode(request, new grpc.Metadata(), (error) => (
+    get(client).navigationService.setMode(request, new grpc.Metadata(), (error) => (
```

Comment:
naive question: what is the purpose of `get`? why is `client.navigationService.setMode` not enough?

Reply:
Ah yeah, this is just a little refactoring using [svelte stores](https://svelte.dev/docs/svelte-store). I've created a new store to access the `client` globally throughout the app (which is done with this `get()`) method. Why do this? Because otherwise we have to pass `client` to any component or function that makes an api call and end up bloating it as the number of components grow. Only the navigation component is making use of this at the moment, but I'll introduce later refactors that'll make a lot of components that previously looked like `<Navigation {client} />` just look like `<Navigation>` and a lot of api methods that looked like `setWaypoint(client, name, location)` look like `setWaypoint(client, name, location)`.